### DISCUSSION TITLE
OpenDSSDirect installation error

### DISCUSSION THREAD

**Discussion by pedroribo** [dss-extensions/dss-extensions#45] [Q&A] [answered]
Hello!
I'm on Windows and was trying to install the OpenDSSDirect package on Anaconda, but got an error code mentioning that no matching distribution was found for dss-python-backend==0.14.5. 

The command I used on Anaconda Prompt to install the package was:

`pip install "OpenDSSDirect.py[extras]"`

which includes the installation of AltDSS-Python, Pandas, NetworkX, SciPy, matplotlib, besides DSS-Python.

The error code in question is the following: 

"ERROR: Could not find a version that satisfies the requirement dss-python-backend==0.14.5 (from dss-python==0.15.7->OpenDSSDirect.py[extras]) (from versions: none)
ERROR: No matching distribution found for dss-python-backend==0.14.5 (from dss-python==0.15.7->OpenDSSDirect.py[extras])"

Is there another required package that I need to install before installing dss-python-backend?

**Comment by PMeira** [comment]
Hi, @pedroribo,
Would you mind listing the output from `pip --version`? I can try to check what's happening this Monday.

**Reply by pedroribo** [reply]
Hi, Paulo!
The output is: 

pip 19.2.3 from c:\users\pedro\appdata\local\programs\python\python37\lib\site-packages\pip (python 3.7)

**Reply by pedroribo** [reply, marked as answer]
I think the error was related to pip indeed. I updated it to a newer version (pip 24.2) and it worked! Thanks!

**Reply by PMeira** [reply]
Nice, thanks for the update!

Sometimes either pip or the wheel package need to be updated to be able to new features and package metadata. That probably explains it. I think it's the first time I see issues on Windows though, typically we see that more on Linux due to the various distributions.

Both pip and the PyPI infra had some major updates starting in 20.x, including a new resolver, so it's good to update in general.

Regarding Python 3.7: while we will support it here due to the number of downloads/users, it's reached EOL -- nice graph at https://devguide.python.org/versions/ about this. Major packages like NumPy have long dropped it too. For DSS-Extensions, we'll reevaluate it in around 6 months, probably dropping both 3.7 and 3.8 then.  Most users are on 3.10 and 3.11 nowadays, as seen on https://pypistats.org/packages/dss-python-backend

**Reply by pedroribo** [reply]
Thanks for the heads-up about Python 3.7! I'll be updating it too.
